src/backend/dx12/src/lib.rs

```diff
@@ -752,6 +752,19 @@ impl hal::Instance for Instance {
                     mem::size_of::<d3d12::D3D12_FEATURE_DATA_ARCHITECTURE>() as _)
             });
 
+            let depth_bounds_test_supported = {
+                let mut features2: d3d12::D3D12_FEATURE_DATA_D3D12_OPTIONS2 = unsafe { mem::zeroed() };
+                if winerror::S_OK == unsafe {
```

Comment:
let's move the function all out of the condition for clarity, we'll then be able to combine the check into `error == winerror::S_OK && features2.DepthBoundsTestSupported != 0`. Although, it depends on what other `winerror` things are expected here.